pkg/cloud/packet/client.go

```diff
@@ -144,23 +147,28 @@ func (p *Client) NewDevice(ctx context.Context, req CreateDeviceRequest) (*packn
 		facility = req.MachineScope.PacketMachine.Spec.Facility
 	}
 
-	serverCreateOpts := &packngo.DeviceCreateRequest{
-		Hostname:      req.MachineScope.Name(),
-		ProjectID:     req.MachineScope.PacketCluster.Spec.ProjectID,
-		Facility:      []string{facility},
-		BillingCycle:  req.MachineScope.PacketMachine.Spec.BillingCycle,
-		Plan:          req.MachineScope.PacketMachine.Spec.MachineType,
-		OS:            req.MachineScope.PacketMachine.Spec.OS,
-		IPXEScriptURL: req.MachineScope.PacketMachine.Spec.IPXEUrl,
-		Tags:          tags,
-		UserData:      userData,
+	hostname := req.MachineScope.Name()
+	serverCreateOpts := metal.CreateDeviceRequest{
+		DeviceCreateInFacilityInput: &metal.DeviceCreateInFacilityInput{
+			Hostname: &hostname,
+			Facility: metal.FacilityInputFacility{
+				String: &facility,
+			},
+			BillingCycle:    &req.MachineScope.PacketMachine.Spec.BillingCycle,
+			Plan:            req.MachineScope.PacketMachine.Spec.MachineType,
+			OperatingSystem: req.MachineScope.PacketMachine.Spec.OS,
+			IpxeScriptUrl:   &req.MachineScope.PacketMachine.Spec.IPXEUrl,
+			Tags:            tags,
+			Userdata:        &userData,
+		},
 	}
 
 	reservationIDs := strings.Split(req.MachineScope.PacketMachine.Spec.HardwareReservationID, ",")
 
 	// If there are no reservationIDs to process, go ahead and return early
-	if len(reservationIDs) == 0 {
-		dev, _, err := p.Devices.Create(serverCreateOpts)
+	if len(reservationIDs) <= 1 {
```

Comment:
`strings.Split` returns an array with length 1 if given an empty string and a non-empty separator, so the code inside this block was not being hit before.  With `packngo`, all device creation was being handled in the for loop on line 180 below.  With `metal-go`, that call fails because we are providing a pointer to `resID`, so the empty `resID` is included in the create request.